superdesk/io/format_document_for_preview.py

```diff
@@ -0,0 +1,43 @@
+from flask import Blueprint, request, Response
+import superdesk
+from superdesk import get_resource_service
+from superdesk.publish.formatters import get_formatter
+from apps.content_types import apply_schema
+
+
+bp = Blueprint('format_document', __name__)
+
+
+def get_mime_type(formatter_qcode):
+    if(formatter_qcode == 'newsmlg2'):
+        return 'text/xml'
+    else:
+        return 'application/json'
+
+
+@bp.route('/format-document-for-preview/', methods=['GET', 'OPTIONS'])
+def format_document():
+
+    document_id = request.args.get('document_id')
+    subscriber_id = request.args.get('subscriber_id')
+    formatter_qcode = request.args.get('formatter')
+
+    subscriber = get_resource_service('subscribers').find_one(req=None, _id=subscriber_id)
+    doc = get_resource_service('archive').find_one(req=None, _id=document_id)
+
+    formatter = get_formatter(formatter_qcode, doc)
+    formatted_docs = formatter.format(article=apply_schema(doc),
```

Comment:
I thought this will make a request to that `preview_endpoint_url` and return the response, not sure why it's doing the internal formatting?

Reply:
It needs to return ninjs/newsml or whatever format is configured for that destination. How would I do it differently?

Reply:
I thought that feature was about rendering html, not about those standard formats

Reply:
The way I saw the problem is that clients want to see how would an article look like when published without actually publishing it. So I went to look at what is happening when actually publishing. I found that we format the item and post it to configured API, so I did the same only without marking it as published and using a different endpoint. It allows clients to reuse their already set up `standard format -> HTML` renderers and provides a realistic production-like preview since the code is going through the same paths.

Do you think it should be changed?

Reply:
that makes sense, but I was expecting this to post the formatted item to configured url and return response from the external system

Reply:
Hey, 

this is exactly what we need for web preview with Publisher. It shouldn't try to return html as you don't know what is the real template. In targeted publishing preview you select website and section and with this data you can see real preview on it's final destination. 

When it can be merged?

Reply:
All we need is an endpoint where I could send item guid and get back a package in same format that is sent when published. 
I get the item like that: `extensionPoints.register('authoring:publish', TargetedPublishing, {session: session, config: config}, ['item']);`

Reply:
@tomaszrondio that's what `/format-document-for-preview/` does except that you also need to send `subscriber_id` and `formatter` in addition to guid. That's because when the item is published, it's published to multiple places in different formats, so you need to select the one you want.

You can check out the PR locally and see if the endpoint behaves like you want. Let me know if it doesn't.